### PR TITLE
Add support for 'docker cp' to write to stdout

### LINKED ISSUE #10805: docker cp should be able write tar to stdout
It looks like the API for Docker cp (`/containers/:id/copy`) natively writes tar format, but I don't see the ability to get the raw tar stream with the docker binary.  It would be great if `docker cp` (or a new command) would stream the tar format directly to stdout.  Similar to what `tar -cf - dir` would do.

We use docker to produce build artifacts that get uploaded to a continuous deployment system and it'd be great to just be able to do this `docker cp container:/output | curl -XPUT -d @- URL`

I see docker export, but that looks like it can only export the entire container.

### PR DESCRIPTION
**Pull request by duglin**
Closes #10805

Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

File: integration-cli/docker_cli_cp_test.go [hunk 1]
```diff
@@ -528,3 +529,30 @@ func TestCpToDot(t *testing.T) {
...
+	out, _, err = dockerCmd(t, "wait", cID)
+	if err != nil || stripTrailingCharacters(out) != "0" {
+		t.Fatalf("failed to set up container:%s\n%s", out, err)
+	}
+
+	out, _, err = runCommandPipelineWithOutput(
+		exec.Command(dockerBinary, "cp", cID+":/test", "-"),
+		exec.Command("tar", "-vtf", "-"))
+	if err != nil {
+		t.Fatalf("Failed to run cp: %s", err)
```
Comment (ahmetalpbalkan) [inline integration-cli/docker_cli_cp_test.go]: tar may also fail here

**Comment by ahmetalpbalkan** [comment]
LGTM

File: integration-cli/docker_cli_cp_test.go (hunk 1 above)
Comment (duglin) [inline integration-cli/docker_cli_cp_test.go]: yup and I would expect either err to be not-nil or the output to be missing what I'm looking for.

**Comment by crosbymichael** [comment]
LGTM

File: integration-cli/docker_cli_cp_test.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline integration-cli/docker_cli_cp_test.go]: I thought "failed to run commands" might be a more comprehensive msg here. No big deal though.

Comment (duglin) [inline integration-cli/docker_cli_cp_test.go]: Ah, I didn't realize that's what you meant.  Sorry.  Next PR, if I remember, I'll sneak it in  :-)

Comment (duglin) [inline integration-cli/docker_cli_cp_test.go]: fixed - for some reason I thought it was already merged.

**Comment by duglin** [comment]
ping @moxiegirl @fredlf

**Comment by SvenDowideit** [comment]
LGTM - @fredlf @jamtur01 @spf13 @moxiegirl

**Comment by jamtur01** [comment]
LGTM

**Pull request merged**